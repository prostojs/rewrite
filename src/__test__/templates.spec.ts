import { ProstoRewrite } from ".."
import { readFileSync, existsSync, unlinkSync } from 'fs'
import Path from 'path'

describe('templates', () => {
    beforeAll(async () => {
        safeUnlink(p('./templates-out/test.html'))
        safeUnlink(p('./templates-out/test.js'))
        safeUnlink(p('./templates-out/rename.json'))
        const rw = new ProstoRewrite()
        await rw.rewriteDir({
            baseDir: p('./templates'),
            output: p('./templates-out'),
            renameFile: name => {
                console.log(name)
                return name.startsWith('__') ? name.slice(2) : name
            },
        }, {
            array: ['item1', 'item2'],
            script: 'http://test',
            path: false,
        })
    })
    it('must rewrite dir', () => {
        expect(existsSync(p('./templates-out/test.html'))).toBeTruthy()
        expect(existsSync(p('./templates-out/test.js'))).toBeTruthy();
        expect(existsSync(p('./templates-out/rename.json'))).toBeTruthy();
        expect(readFileSync(p('./templates-out/test.html')).toString()).toMatchSnapshot()
        expect(
            readFileSync(p('./templates-out/test.js')).toString(),
        ).toMatchSnapshot();
        expect(
            readFileSync(p('./templates-out/rename.json')).toString(),
        ).toMatchSnapshot();
    })
})

function p(path: string) {
    return Path.join(process.cwd(), 'src', '__test__', path)
}

function safeUnlink(file: string) {
    try {
        unlinkSync(file)
    } catch {}
}
