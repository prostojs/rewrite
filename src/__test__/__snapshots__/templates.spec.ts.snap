// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates must rewrite dir 1`] = `
"<!DOCTYPE html>
<html>
    <head>
        <script src="http://test">
        </script>
    </head>
    <body>
        <div >
            item1
        </div>
    <div >
            item2
        </div>
    </body>
</html>"
`;

exports[`templates must rewrite dir 2`] = `
"const val = 'http://test'
"
`;

exports[`templates must rewrite dir 3`] = `
"{
    "toBeRenamed": true
}"
`;
